packages/teams-js/src/internal/utils.ts

```diff
@@ -416,6 +416,19 @@ export function validateId(id: string, errorToThrow?: Error): void {
   }
 }
 
+export function validateUrl(url: string, errorToThrow?: Error): void {
```

Comment:
It seems odd to have this function and isValidHttpsUrl both exist in the utils file, will cause confusion for people deciding which one to use. Seems like there are a couple of options:

1) You could update `isValidHttpsURL` in this file to have this new logic that you're introducing here so everyone gets to use the new logic, or
2) You could make two separate functions but share the actual validation code between them both.

Would be good to not fork our Https URL validation code and have two different validation functions that validate differently.